controllers/api/api_trusted_controller.py

```diff
@@ -108,11 +109,16 @@ def _process_request(self, request, event_key):
 
             if (not match.time or match.time == "") and match.time_string:
                 # We can calculate the real time from the time string
-                logging.debug("Calculating time!")
-                MatchHelper.add_match_times(event, [match])
-
+                needs_time.append(match)
             matches.append(match)
 
+        if needs_time:
```

Comment:
Why not keep this in the original for loop? That way one match can fail, but the others can succeed.